chipflow_lib/platforms/sim.py

```diff
@@ -3,36 +3,175 @@
 import logging
 import os
 import sys
+
+from dataclasses import dataclass
+from enum import StrEnum
 from pathlib import Path
+from pprint import pformat
+from typing import Dict, List, Optional, Type
 
-from amaranth import Module, ClockDomain, ClockSignal, ResetSignal
-from amaranth.lib import io
+from amaranth import Module, ClockSignal, ResetSignal, ClockDomain
+from amaranth.lib import io, wiring
 from amaranth.back import rtlil  # type: ignore[reportAttributeAccessIssue]
 from amaranth.hdl._ir import PortDirection
 from amaranth.lib.cdc import FFSynchronizer
+from jinja2 import Environment, PackageLoader, select_autoescape
+from pydantic import BaseModel
 
-from ._utils import load_pinlock
+from .. import ChipFlowError, _ensure_chipflow_root
+from ._signatures import (
+        I2CSignature, GPIOSignature, UARTSignature, SPISignature, QSPIFlashSignature,
```

Comment:
is there any way we could avoid this hardcoded set here?

Reply:
These are only the signatures used by the built in models.
I do still need to add an easy way for a user to add more models. most of the logic is there, but needs a bit of api.

Reply:
fair enough!